mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -338,13 +341,27 @@ def _get_workspace_id(self, run_id):
                 "current user. No run link will be recorded for the model version."
             )
             return None
-        if _DATABRICKS_ORG_ID_HEADER not in response.headers:
+        return response
+
+    def _get_workspace_id(self, get_run_response_proto):
+        if get_run_response_proto is None or _DATABRICKS_ORG_ID_HEADER not in get_run_response_proto.headers:
             _logger.warning(
                 "Unable to get model version source run's workspace ID from request headers. "
                 "No run link will be recorded for the model version"
             )
             return None
-        return response.headers[_DATABRICKS_ORG_ID_HEADER]
+        return get_run_response_proto.headers[_DATABRICKS_ORG_ID_HEADER]
+
+    def _get_notebook_id(self, get_run_response_proto):
+        run = Run.from_proto(get_run_response_proto.run)
+        params = run.data.params
+        notebook_id = params.get(_DATABRICKS_NOTEBOOK_ID_KEY, None)
+        if notebook_id is None:
+            _logger.warning(
```

Comment:
No need to log a warning in this case, it's probably pretty common (like any time a model version is created in a job)

Reply:
Or from a model trained locally etc